docs/src/operate/troubleshooting/tedge-diag.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Diagnostic Collection
+tags: [Operate]
+sidebar_position: 1
+description: How to collect logs and system diagnostics using plugin-based automation
+---
+
+%%te%% helps you collect useful diagnostic information about your device and services by executing diagnostic plugins.
+These diagnostics are bundled into a single compressed archive (`.tar.gz`) for easy analysis or sharing with support.
+
+This guide explains how to use the diagnostic collection tool and how plugins work.
+
+## Getting Started
+
+To collect a snapshot of diagnostic data, run:
+
+```sh
+tedge diag collect
+```
+
+This command performs the following steps:
+* Discovers all available diagnostic plugins (default location: `/usr/share/tedge/diag-plugins`)
+* Executes each plugin in sequence
+* Captures their logs and outputs
+* Bundles all collected data into a .`tar.gz` archive (e.g., `/tmp/tedge-diag-YYYY-MM-DD_HH-MM-SS.tar.gz`)
+
+All directory paths and the package name can be customized using command-line arguments or the `tedge config` command.
+
+To see all available options, run:
+
+```sh
+tedge diag collect --help
+```
+
+## Diagnostic Plugins
```

Comment:
This section gives the required content but in an order that can be improved.

- What is a diagnostic plugin
- Where to find them
- What is provided out-of-the-box
- What has to be done to add diagnostic plugins
- What is the contract between the runner and the plugins

Reply:
Nicely done. I would just move the "Disabling plugins" section after the list of "Predefined plugins". The need to disable plugins comes only once aware of what is provided out-of-the-box.